api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -194,6 +194,10 @@ type NodePoolSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Arch is immutable"
 	// +optional
 	Arch string `json:"arch,omitempty"`
+
+	// FailureDomain is the preferred failure domain for the NodePool.
+	// +optional
+	FailureDomain *string `json:"failureDomain,omitempty"`
```

Comment:
Need this be a pointer? Given the empty string is not a valid failure domain name, I think it probably doesn't